android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java

```diff
@@ -105,22 +108,23 @@ public void onReceived() throws InvalidNotificationException {
 
         if (channelId != null) {
             notificationId = channelId.hashCode();
-            Object objCount = channelIdToNotificationCount.get(channelId);
-            Integer count = 1;
-            if (objCount != null) {
-                count = (Integer)objCount + 1;
-            }
-            channelIdToNotificationCount.put(channelId, count);
 
-            Object bundleArray = channelIdToNotification.get(channelId);
-            List list = null;
-            if (bundleArray == null) {
-                list = Collections.synchronizedList(new ArrayList(0));
-            } else {
-                list = Collections.synchronizedList((List)bundleArray);
+            synchronized (channelIdToNotificationCount) {
+                Integer count = channelIdToNotificationCount.get(channelId);
+                if (count == null) {
+                    count = 0;
+                }
+
+                channelIdToNotificationCount.put(channelId, count + 1);
             }
-            synchronized (list) {
-                if (!"clear".equals(type)) {
+
+            synchronized (channelIdToNotification) {
+                List<Bundle> list = channelIdToNotification.get(channelId);
+                if (list == null) {
+                    list = Collections.synchronizedList(new ArrayList(0));
+                }
+
+                if (PUSH_TYPE_MESSAGE.equals(type)) {
```

Comment:
I'm guessing that it's intentional to change from anything but clear, to only message, but can you confirm this change is on purpose? I can only ask, as I'm not aware of the differences between the three types.

Reply:
Yeah, it's intentional. Previously, the two types of notifications were just `message` (show a visible notification for a new message) and `clear` (clear a previous notification). This is adding support for the new `update_badge` notification type that isn't supposed to show a visible notification. `update_badge` would ideally update the badge on the app icon, but that doesn't seem possible on Android, so we're just making Android devices ignore it silently instead of showing a blank notification which is what it would do currently.

This block is adding an entry to `channelIdToNotification[channelId]` for each visible notification, so we only want to do that for `message` notifications. The old logic just had the logic reversed since it was adding an entry for anything that wasn't a `clear`.